### PR TITLE
Add integration test for amp-analytics various triggers.

### PR DESCRIPTION
**Pull request by lannka**
part of #19706

### CODE REVIEW

File: test/integration/test-amp-analytics.js [hunk 1]
```diff
@@ -55,6 +55,54 @@ describe.configure().skipIfPropertiesObfuscated().run('amp' +
...
+
+    beforeEach(() => {
+      browser = new BrowserHandle(env.win);
+    });
+
+    it('should send request', () => {
+      const reqPromise = RequestBank.withdraw().then(req => {
+        expect(req.url).to.equal('/?f=hello%20world&b=2');
+      });
+      browser.click('a');
```
Comment (zhouyx) [inline test/integration/test-amp-analytics.js]: Looks good. But is it possible to test that the request is sent after the click, or not if we click on other element? With fake clock maybe?

Comment (lannka) [reply test/integration/test-amp-analytics.js]: For such an integration test fake clock might not be feasible. 
I thought about delaying the click for 1s, and then look at the req timestamp to make sure it's after the click. But I feel not worth the wait, as it's a very unlikely regression (fire a click trigger without click).

Such an approach could be used for "element visible" trigger, where such a regression is more likely to happen.

File: testing/test-helper.js [hunk 2]
```diff
@@ -189,6 +189,20 @@ export class RequestBank {
   }
 }
 
+export class BrowserHandle {
```
Comment (calebcordry) [inline testing/test-helper.js]: I want to bikeshed this name :) `BrowserInterface`? `BrowserController`?

**Comment by lannka** [comment]
hi all pls take another look. I've added more tests to cover various triggers in the same PR.

File: test/integration/test-amp-analytics.js [hunk 3]
```diff
@@ -66,7 +66,216 @@ describe.configure().skipIfPropertiesObfuscated().run('amp' +
...
+              "extraUrlParams": {
+                "scrollTop": "\${scrollTop}",
+                "scrollHeight": "\${scrollHeight}"
+              }
+            }
+          }
+        }
+        </script>
+      </amp-analytics>
+      <div class="block" style="height: 100vh; background: red">
```
Comment (zhouyx) [inline test/integration/test-amp-analytics.js]: This is for debugging?

File: test/integration/test-amp-analytics.js [hunk 4]
```diff
@@ -66,7 +66,216 @@ describe.configure().skipIfPropertiesObfuscated().run('amp' +
...
+  }, env => {
+    let browser;
+
+    beforeEach(() => {
+      browser = new BrowserController(env.win);
+    });
+
+    it('should trigger on scroll', () => {
+      const reqPromise = RequestBank.withdraw().then(req => {
+        expect(req.url).to.equal('/?scrollTop=75&scrollHeight=300');
```
Comment (zhouyx) [inline test/integration/test-amp-analytics.js]: where do we specify the 300? will the scrollHeight=300 be flaky?

File: test/integration/test-amp-analytics.js [hunk 5]
```diff
@@ -66,7 +66,216 @@ describe.configure().skipIfPropertiesObfuscated().run('amp' +
...
+              }
+            }
+          }
+        }
+        </script>
+      </amp-analytics>
+      <div class="block" style="height: 100vh; background: red">
+        1st viewport
+      </div>
+      <amp-img layout="fixed" width="300" height="150"
```
Comment (zhouyx) [inline test/integration/test-amp-analytics.js]: same question, can we always be sure that the image can be 50% visible, that the window size is twice more than image's size.

File: test/integration/test-amp-analytics.js [hunk 6]
```diff
@@ -66,7 +66,216 @@ describe.configure().skipIfPropertiesObfuscated().run('amp' +
...
+          }
+        }
+        </script>
+      </amp-analytics>
+      `,
+    extensions: ['amp-analytics'],
+  }, () => {
+    it('should trigger 1s after amp-analytics starts', () => {
+      const startTime = Date.now();
+      return RequestBank.withdraw().then(req => {
```
Comment (zhouyx) [inline test/integration/test-amp-analytics.js]: You also set the `maxTimerLength` to be 2. Maybe worth waiting another 1s and make sure no additional request has been sent.

File: test/integration/test-amp-analytics.js [hunk 7]
```diff
@@ -273,58 +506,7 @@ describe.configure().skipIfPropertiesObfuscated().run('amp' +
     });
   });
 
-  describes.integration('amp-analytics useBody with extraUrlParamsReplaceMap', {
```
Comment (zhouyx) [inline test/integration/test-amp-analytics.js]: why are we removing this?

**Comment by calebcordry** [comment]
Just a general comment, are we worried about introducing guaranteed latency into these tests (ie wait for 2 seconds)? My fear is that if we have a lot of these it becomes very slow.

File: test/integration/test-amp-analytics.js (hunk 7 above)
Comment (calebcordry) [reply test/integration/test-amp-analytics.js]: +1

**Pull request merged**